barcode/templates/barcode_client.html

```diff
@@ -0,0 +1,246 @@
+      preventDefault: false,
+      stopPropagation: false,
+      onComplete: function(barcode, qty) {
+        barcode = ('' + barcode).trim();
+
+        // Remove non-word characters bracketing the barcode.
+        // Some scanners add these but we don't want them.
+        var first = barcode.substr(0, 1),
+            last = barcode.substr(barcode.length - 1);
+        if (first === last && last.match(/\W/g)) {
+          barcode = barcode.substr(1, barcode.length - 2);
+        }
+
+        if (scannerDetectionUUIDHandlers.length > 0) {
+          var uuidMatch = barcode.match(RE_UUID);
+          if (uuidMatch) {
+            $.each(scannerDetectionUUIDHandlers, function(i, handler) {
+              handler({barcode: barcode, uuid: uuidMatch[1]});
+            });
+          }
+        }
+
+        if (scannerDetectionBadgeNumHandlers.length > 0) {
+          getBadgeNumFromBarcode(barcode, function (response) {
+            $.each(scannerDetectionBadgeNumHandlers, function(i, handler) {
+              handler(response);
+            });
+          });
+        }
+
+        $.each(scannerDetectionOnCompleteHandlers, function(i, handler) {
+          handler({barcode: barcode});
+        });
+      },
+      onError: function (key) {
+        $.each(scannerDetectionOnErrorHandlers, function(i, handler) {
+          handler(key);
+        });
+      },
+      onReceive: function(event) {
+        $.each(scannerDetectionOnReceiveHandlers, function(i, handler) {
+          handler(event);
+        });
+      }
+    });
+  });
+
+  (function($) {
+    $.fn.barcodeField = function(options) {
+      // Default settings, overridden by options
+      var settings = $.extend({
+        // The target field for auto-fill and focus/blur events, if different
+        // from the element used to initialize the barcodeField.
+        targetField: null,
+
+        // Control how the target field is blurred/focused.
+        blurOnKeys: [],
+        refocusAfterBlurOnKeys: true,  // Refocuses if blurred from blurOnKeys
+        focusOnError: true,  // Always focus, regardless of blurOnKeys
+        focusOnComplete: true,  // Always focus, regardless of blurOnKeys
+
+        // Turn off toastr error messages entirely.
+        showErrorMessage: true,
+
+        // Turn off toastr error messages, but only when target field is hidden.
+        showErrorMessageWhenHidden: true,
+
+        // Determine if we do any special data handling.
+        // detectBadgeNum and detectUUID are mututally exclusive. If both are
+        // true, then only detectBadgeNum is used.
+        detectBadgeNum: false,
+        detectUUID: false,
+
+        // Determine if the target field is auto-filled.
+        autoFill: true,
+
+        // Optional function to determine if target field should actually
+        // be auto-filled. Return true to auto-fill.
+        // Ignored if autoFill is falsey.
+        // function() { return true; }
+        shouldAutoFill: null,
```

Comment:
I was a little confused by this at first -- suggest renaming to `autoFillWhen` and the submit version to `autoSubmitWhen` to make their purpose a little more obvious.

In fact, if you do that, you can even rename `autoFill` to `shouldAutoFill` to differentiate it more from `autoSubmitForm`, which at first seems like its parallel when in fact they are different types of values. If... if that makes sense.